src/redux/rockets/rocketsSlice.js

```diff
@@ -0,0 +1,67 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+export const fetchRockets = createAsyncThunk('rockets/fetchRockets', async () => {
+  try {
+    const response = await axios.get('https://api.spacexdata.com/v3/rockets');
+    return response.data.map((rocket) => ({
+      id: rocket.id,
+      name: rocket.rocket_name,
+      description: rocket.description,
+      flickr_images: rocket.flickr_images[0],
+      reserved: false,
+    }));
+  } catch (error) {
+    throw new Error('Failed to fetch rockets.');
+  }
+});
+
+export const joinRocket = createAsyncThunk(
+  'rockets/joinRocket',
+  async (rocketId) => rocketId,
+);
+
+export const leaveRocket = createAsyncThunk(
+  'rockets/leaveRocket',
+  async (rocketId) => rocketId,
+);
```

Comment:
- Optional.
`createAsyncThunk` is specifically designed for handling async actions like API calls. In the case of `joinRocket` and `leaveRocket`, which are simple actions, it is not necessary to use `createAsyncThunk`. Instead, you can define them directly within the reducers as regular action creators. By doing so, you will eliminate the overhead of `createAsyncThunk`, which is more suitable for handling asynchronous actions. This will simplify the code and make it easier to reason about.